src/compiler/emitter.ts

```diff
@@ -300,14 +300,29 @@ namespace ts {
                             continue;
                         }
 
-                        writeLines(helper.text);
+                        writeHelper(writeLines, helper.text, makeUniqueName);
                         helpersEmitted = true;
                     }
                 }
             }
 
             return helpersEmitted;
         }
+
+        function writeHelper(writer: (text: string) => void, helper: EmitHelper["text"], makeUniqueName: UniqueNameHandler) {
+            if (typeof helper === "string") {
+                writer(helper);
+            }
+            else {
+                writer(helper({

```

Comment:
Is there a reason you allocate an object just to pass a callback? Could this not just be the callback itself?